src/node.rs

```diff
@@ -173,6 +196,19 @@ impl<T> MutableNode for Node<T> {
         self.children.push(child);
     }
 
+    fn replace_child(&mut self, old: NodeId, new: NodeId) {
+        let mut index = 0;
+
+        for (i, id) in self.children().iter().enumerate()
+        {
+            if id == &old { index = i; break; }
+        }
```

Comment:
I like the iterator patters very much, no idea, why I did not though about this.